tests/integration/conftest.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Fixtures for Jenkins-agent-k8s-operator charm integration tests."""
+
+import logging
+import secrets
+import typing
+
+import jenkinsapi.jenkins
+import pytest
+import pytest_asyncio
+from juju.action import Action
+from juju.application import Application
+from juju.client._definitions import FullStatus, UnitStatus
+from juju.model import Controller, Model
+from juju.unit import Unit
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+BLOCKED_STATUS_NAME = "blocked"
+
+
+@pytest_asyncio.fixture(scope="module", name="charm")
+async def charm_fixture(request: pytest.FixtureRequest, ops_test: OpsTest) -> str:
+    """The path to charm."""
+    charm = request.config.getoption("--charm-file")
+    if not charm:
+        charm = await ops_test.build_charm(".")
+    else:
+        charm = f"./{charm}"
+
+    return charm
+
+
+@pytest.fixture(scope="module", name="model")
+def model_fixture(ops_test: OpsTest) -> Model:
+    """The testing model."""
+    assert ops_test.model
+    return ops_test.model
+
+
+@pytest.fixture(scope="module", name="num_agents")
+def num_agents_fixture() -> int:
+    """The number of agents to deploy."""
+    return 1
```

Comment:
I'm curios about the reasoning for making this a fixture? Fixtures are great, they do come with an overhead of a layer of abstraction so there needs to be a counter balancing benefit that makes them worth having

Reply:
I agree on this one that it makes more sense for this to be a constant, I updated it.